<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1150</width>
    <height>1048</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1150</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>ChargeGuru 0.0.1</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="gbChargingParameters">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Charging Parameters</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Battery type:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cbBatteryType">
         <property name="currentText">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Charging mode:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cbChargingMode"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Cell count:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lbCellCount">
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QSlider" name="sbCellCount">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>6</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="invertedAppearance">
          <bool>false</bool>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBothSides</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Charge current:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbChargeCurrent">
         <property name="suffix">
          <string>mA</string>
         </property>
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>6000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Discharge current:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbDischargeCurrent">
         <property name="suffix">
          <string>mA</string>
         </property>
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>2000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Cell discharge voltage:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbCellDischargeVoltage">
         <property name="suffix">
          <string>mV</string>
         </property>
         <property name="minimum">
          <number>3000</number>
         </property>
         <property name="maximum">
          <number>3300</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>End voltage:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbEndVoltage">
         <property name="suffix">
          <string>mV</string>
         </property>
         <property name="minimum">
          <number>4100</number>
         </property>
         <property name="maximum">
          <number>4250</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Repeak count:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbRepeakCount">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Cycle count:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbCycleCount">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="btStartCharging">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btStopCharging">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" rowspan="2">
     <widget class="QGroupBox" name="gbChargingInfo">
      <property name="title">
       <string>Charging Info</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="2">
        <widget class="QTableWidget" name="tbCells">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>52</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QTableWidget, QHeaderView::section, QHeaderView, QTableCornerButton::section {
	background: transparent;
	border: none;
}

QHeaderView {
	padding: 0 0 7px 0;
	border-bottom: 1px solid rgba(0, 0, 0, 0.3);
}
</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="lineWidth">
          <number>0</number>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="sizeAdjustPolicy">
          <enum>QAbstractScrollArea::AdjustIgnored</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="gridStyle">
          <enum>Qt::NoPen</enum>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <property name="rowCount">
          <number>1</number>
         </property>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>Voltage</string>
          </property>
         </row>
         <column>
          <property name="text">
           <string>Cell 1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 3</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 4</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 5</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 6</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 7</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Cell 8</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="1">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="2">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="3">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="4">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="5">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="6">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
         <item row="0" column="7">
          <property name="text">
           <string>0.000V</string>
          </property>
          <property name="textAlignment">
           <set>AlignCenter</set>
          </property>
          <property name="flags">
           <set>ItemIsEnabled</set>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QCheckBox" name="ckChartCurrent">
           <property name="text">
            <string>Current</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="ckChartVoltage">
           <property name="text">
            <string>Voltage</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="ckChartCellsVoltage">
           <property name="text">
            <string>Cells Voltage</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="ckChartCapacity">
           <property name="text">
            <string>Capacity</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="ckChartTemp">
           <property name="text">
            <string>Temperature</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>Time:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeTime">
             <property name="text">
              <string>00:00:00</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>Current:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeCurrent">
             <property name="text">
              <string>0.000 A</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>Voltage:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeVoltage">
             <property name="text">
              <string>0.000 V</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>Capacity:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeCapacity">
             <property name="text">
              <string>0 mAh</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Temp. Ext.:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeTempExt">
             <property name="text">
              <string>0°C</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Temp. Int.:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lbChargeTempInt">
             <property name="text">
              <string>0°C</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,1,1,1,1">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <item>
          <widget class="QtCharts::QChartView" name="ctCurrent">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QtCharts::QChartView" name="ctVoltage">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QtCharts::QChartView" name="ctCellsVoltage">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QtCharts::QChartView" name="ctCapacity">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QtCharts::QChartView" name="ctTemp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="gbSystemSettings">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>System Settings</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QCheckBox" name="ckTimeLimit">
         <property name="text">
          <string>Time limit:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbTimeLimit">
         <property name="suffix">
          <string> min</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>720</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="ckCapacityLimit">
         <property name="text">
          <string>Capacity limit:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbCapacityLimit">
         <property name="suffix">
          <string>mAh</string>
         </property>
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>50000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Temperature cut-off:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbTemperatureLimit">
         <property name="suffix">
          <string>°C</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>80</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Rest time:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbCycleTime">
         <property name="suffix">
          <string> min</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>60</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="ckKeyBuzzer">
         <property name="text">
          <string>Key buzzer</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="ckSystemBuzzer">
         <property name="text">
          <string>System buzzer</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="btSave">
           <property name="text">
            <string>Save</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btLoad">
           <property name="text">
            <string>Load</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QtCharts::QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">QtCharts/QChartView&gt;
#include &lt;QtCharts/chartsnamespace.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
